CALCULATOR_UPDATED.c

```diff
@@ -0,0 +1,164 @@
+    while (user_input[iterator_input] != '\0') 
+    {
+        if (user_input[iterator_input] != ' ') 
+        {
+            user_input[iterator_current++] = user_input[iterator_input];
+        }
+        iterator_input++;
+    }
+    user_input[iterator_current] = '\0';
+}
+
+
+bool isDigit(char ch) 
+{
+    if(ch>='0' && ch<='9')
+    {
+        return true;
+    }
+    return false;
+}
+
+
+bool isOperator(char ch) 
+{
+    for (int i=0; OPERATORS[i]!='\0';i++)
+    {
+        if (ch==OPERATORS[i]) 
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+
+void checkInput(char user_input[]) 
+{
+    for (int i = 0; user_input[i] != '\0'; i++) 
+    {
+        if (!isDigit(user_input[i]) && !isOperator(user_input[i])) 
+        {
+            printf("Error: Invalid character in expression.\n");
+            exit(0);
+        }
+    }
+}
+
+
+int checkPrecedence(char op)
+{
+    if (op == '*' || op == '/') return 2;
+    if (op == '+' || op == '-') return 1;
+}
+
+
+int computeOpeartions(int operand1, int operand2, char operator) 
+{
+    if (operator == '+')      return operand1 + operand2; 
+    else if (operator == '-') return operand1 - operand2;
+    else if (operator == '*') return operand1 * operand2; 
+    else if (operator == '/') 
+    {
+        if (operand2 != 0) 
+        {
+            return operand1 / operand2; 
+        }
+        else
+        {
+        printf("Error:Division by zero.\n");
+        exit(0); 
+        }
+    }
+}
+
+
+int traverseNumber(char user_input[], int *iterator) 
+{
+    int num = 0;
+    while (isDigit(user_input[*iterator])) 
+    {
+        num = num * 10 + (user_input[*iterator] - '0');
+        (*iterator)++;
+    }
+    return num;
+}
+
+
+void accessOpeartor(int *numbers, int *num_idx, char *operators, int *op_idx)
+{
+    int operand2 = numbers[--(*num_idx)];
+    int operand1 = numbers[--(*num_idx)];
+    char operator = operators[--(*op_idx)];
+    numbers[(*num_idx)++] = computeOpeartions(operand1, operand2, operator);
+}
+
+
+int evaluateExpression(char user_input[])
+{
+    int numbers[100];          
+    char operators[100];      
+    int num_idx = 0;      
+    int op_idx = 0;   
+
+    int iterator = 0;         
+
+    while (user_input[iterator] != '\0') 
+    {
+        if (isDigit(user_input[iterator])) 
+        {
+            numbers[num_idx++] = traverseNumber(user_input, &iterator);
+        } 
+        else if (isOperator(user_input[iterator])) 
+        {
+            while (op_idx > 0 && checkPrecedence(operators[op_idx - 1]) >= checkPrecedence(user_input[iterator]))
+            {
+                accessOpeartor(numbers, &num_idx, operators, &op_idx);
+            }
+            operators[op_idx++] = user_input[iterator];
+            iterator++;
+        } 
+    }
+    while (op_idx > 0) 
+    {
+        accessOpeartor(numbers, &num_idx, operators, &op_idx);
+    }
+    return numbers[0];
+}
+
+
+int main() {

```

Comment:
no need of separate new file, you should edit the same file for fixing review comments and need to push the code, please refer any online source like how to fix the review comments and push the code.